tokio-macros/src/entry.rs

```diff
@@ -437,14 +457,21 @@ fn parse_knobs(mut input: ItemFn, is_test: bool, config: FinalConfig) -> TokenSt
     let body_ident = quote! { body };
     // This explicit `return` is intentional. See tokio-rs/tokio#4636
     let last_block = quote_spanned! {last_stmt_end_span=>
+        #(#checks)*
+        #(#attrs)*
         #[allow(clippy::expect_used, clippy::diverging_sub_expression, clippy::needless_return)]
         {
             return #rt
                 .enable_all()
-                .build()
+                .#build
                 .expect("Failed building the Runtime")
                 .block_on(#body_ident);
         }
+
+        #[allow(unreachable_code)]
+        {
+            panic!("fell through checks")
+        }
```

Comment:
Is this code needed? (since this is after the return statement, I think it will never be executed.)

Reply:
Or, does this work as a safeguard for future macro changes?

Reply:
As far as I can tell, the addition of this panic is the cause of the worse error messages seen in `tests-build/tests/fail/macros_type_mismatch.stderr`. Having reasonably good error messages when used incorrectly is a goal of the macro.

Can you try annotating the fall-through case with `#[cfg(not(tokio_unstable))]` and see if that avoids the regression on the error messages.